src/test/java/uk/ac/ebi/ampt2d/registry/PropertyRegistryServiceApplicationTests.java

```diff
@@ -236,11 +246,45 @@ public void testPaging() throws Exception {
 
         postTestEntity("/properties", content2);
 
-        mockMvc.perform(get("/properties?size=1")).andExpect(status().isOk())
+        mockMvc.perform(get("/properties?size=1").with(oAuthHelper.bearerToken("testUser@gmail.com")))
+                .andExpect(status().isOk())
                 .andExpect(jsonPath("$._embedded.properties.length()").value(1));
-        mockMvc.perform(get("/properties?size=2")).andExpect(status().isOk())
+        mockMvc.perform(get("/properties?size=2").with(oAuthHelper.bearerToken("testUser@gmail.com")))
+                .andExpect(status().isOk())
                 .andExpect(jsonPath("$._embedded.properties.length()").value(2));
 
     }
 
+    @Test
+    public void testAuthorization() throws Exception {
+        // Any url other than root and swagger is Secured
+        mockMvc.perform(get("/phenotypes")).andExpect(status().isUnauthorized());
+        mockMvc.perform(get("/properties")).andExpect(status().isUnauthorized());
+
+        mockMvc.perform(get("/")).andExpect(status().isOk()); // Root is not secured
+        mockMvc.perform(get("/swagger-ui.html")).andExpect(status().isOk()); // Swagger is not secured
+
+        String content = "{\"id\":\"CALL_RATE\"," +
+                "\"type\":\"FLOAT\"," +
+                "\"meaning\":\"CALL_RATE\"," +
+                "\"description\":\"calling rate\"}";
+
+        //POST can be performed  by EDITOR or ADMIN only
+        mockMvc.perform(post("/properties").with(oAuthHelper.bearerToken("testUser@gmail.com"))
+                .content(content)).andExpect(status().isForbidden());
+        mockMvc.perform(post("/properties").with(oAuthHelper.bearerToken("testEditor@gmail.com"))
+                .content(content)).andExpect(status().isCreated());
```

Comment:
For completeness, I would also add test cases for other request methods and roles combinations.